src/Fixer/Phpdoc/PhpdocFullyQualifiedTypesFixer.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Phpdoc;
+
+use PhpCsFixer\AbstractPhpdocTypesFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\TypeAnalysis;
+use PhpCsFixer\Tokenizer\Resolver\TypeShortNameResolver;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class PhpdocFullyQualifiedTypesFixer extends AbstractPhpdocTypesFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Transforms imported FQCN in docblocks to short version.',
+            [new CodeSample('<?php
+
+use Foo\Bar;
+
+/**
+ * @param \Foo\Bar $foo
+ */
+function foo($foo) {}
+')]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before GeneralPhpdocAnnotationRemoveFixer, NoBlankLinesAfterPhpdocFixer, NoEmptyPhpdocFixer, NoSuperfluousPhpdocTagsFixer, PhpdocAddMissingParamAnnotationFixer, PhpdocAlignFixer, PhpdocAlignFixer, PhpdocAlignFixer, PhpdocAnnotationWithoutDotFixer, PhpdocInlineTagFixer, PhpdocLineSpanFixer, PhpdocNoAccessFixer, PhpdocNoAliasTagFixer, PhpdocNoEmptyReturnFixer, PhpdocNoPackageFixer, PhpdocNoUselessInheritdocFixer, PhpdocOrderByValueFixer, PhpdocOrderFixer, PhpdocReturnSelfReferenceFixer, PhpdocSeparationFixer, PhpdocSingleLineVarSpacingFixer, PhpdocSummaryFixer, PhpdocToParamTypeFixer, PhpdocToReturnTypeFixer, PhpdocTrimConsecutiveBlankLineSeparationFixer, PhpdocTrimFixer, PhpdocTypesOrderFixer, PhpdocVarAnnotationCorrectOrderFixer, PhpdocVarWithoutNameFixer.
+     * Must run after CommentToPhpdocFixer, PhpdocScalarFixer.
+     */
+    public function getPriority()
+    {
+        /*
+         * Should be run before all other docblock fixers apart from the
+         * phpdoc_to_comment and phpdoc_indent fixer to make sure all fixers
+         * apply correct indentation to new code they add. This should run
+         * before alignment of params is done since this fixer might change
+         * the type and thereby un-aligning the params.
+         */
+        return 14;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function normalize(Tokens $tokens, $type)
+    {
+        if (0 !== strpos($type, '\\')) {
+            return $type;
+        }
+
+        // Dummy start and end indexes, we only need isReservedType result
+        if ((new TypeAnalysis(substr($type, 1), 1, 2))->isReservedType()) {
```

Comment:
since https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7188/files#diff-9eb8d1884736f293d0a3ea5127a6713fea7a744e293f41b48b263e910297a4bdR66 the `1, 2` dummy values are not needed